keras_nlp/models/xlnet/xlnet_backbone.py

```diff
@@ -0,0 +1,217 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""XLNet backbone model."""
+
+
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.backend import keras
+from keras_nlp.models.backbone import Backbone
+from keras_nlp.models.xlnet.xlnet_content_and_query_embedding import (
+    ContentAndQueryEmbedding,
+)
+from keras_nlp.models.xlnet.xlnet_encoder import XLNetAttentionMaskLayer
+from keras_nlp.models.xlnet.xlnet_encoder import XLNetEncoder
+from keras_nlp.models.xlnet.xlnet_encoder import XLNetSegmentMatrixLayer
+
+
+@keras_nlp_export("keras_nlp.models.XLNetBackbone")
+class XLNetBackbone(Backbone):
+    """XLNet encoder network.
+
+    This class implements a XLNet Transformer.
+
+    The default constructor gives a fully customizable, randomly initialized
+    XLNet encoder with any number of layers, heads, and embedding dimensions.
+    To load preset architectures and weights, use the `from_preset` constructor.
+
+    Disclaimer: Pre-trained models are provided on an "as is" basis, without
+    warranties or conditions of any kind.
+
+    Attributes:
+        vocabulary_size: int. The size of the token vocabulary.
+        num_layers: int. The number of transformer encoder layers.
+        num_heads: int, the number of heads in the
+            `keras.layers.TwoStreamRelativeAttention` layer.
+        hidden_dim: int, the size hidden states.
+        intermediate_dim: int, the hidden size of feedforward network.
+        dropout: float, defaults to 0.0 the dropout value, shared by
+            `keras.layers.TwoStreamRelativeAttention` and feedforward network.
+        activation: string or `keras.activations`, defaults to "gelu". the
+            activation function of feedforward network.
+        kernel_initializer_range: int, defaults to 0.02. The kernel initializer
+            range for the dense and relative attention layers.
+        bias_initializer: string or `keras.initializers` initializer,
+            defaults to "zeros". The bias initializer for
+            the dense and multiheaded relative attention layers.
+        layer_norm_epsilon: float, defaults to 1e-12. The epsilon value in layer
+            normalization components.
+        **kwargs: other keyword arguments.
```

Comment:
this is an anti-pattern we should delete from our repo. no need to document this, it is true of all objects in the library really.